cve_bin_tool/output_engine/util.py

```diff
@@ -25,9 +26,15 @@ def generate_filename(extension: str) -> str:
         str: unique generated filename
     """
     now = datetime.now().strftime("%Y-%m-%d.%H-%M-%S")
-    filename = os.path.abspath(
-        os.path.join(os.getcwd(), f"output.cve-bin-tool.{now}.{extension}")
-    )
+    if not append_flag:
+        filename = os.path.abspath(
+            os.path.join(os.getcwd(), f"output.cve-bin-tool.{now}.{extension}")
+        )
+    else:
+        filename = os.path.abspath(
+            os.path.join(os.getcwd(), f"intermediate.cve-bin-tool.{now}.{extension}")
+        )
```

Comment:
if branches are nearly identical, might read easier if you just replace substring if needed:
```suggestion
    filename = os.path.abspath(
        os.path.join(os.getcwd(), f"output.cve-bin-tool.{now}.{extension}")
    )
    if append_flag:
        filename = filename.replace("output", "intermediate")
```

Reply:
Thanks, @Molkree  I have made all the changes your suggested. Do let me know if I missed something.